url/resources/toascii.json

```diff
@@ -198,5 +199,174 @@
   {
     "input": ">\u00AD\u0338",
     "output": "xn--hdh"
+  },
+  "Tests below are from WebKit (fast/url/idna2003.html & fast/url/idna2008.html; contributed by Chris Weber back in 2011).",
+  {
+    "input": "faß.de",
+    "output": "xn--fa-hia.de"
+  },
+  {
+    "input": "βόλος.com",
+    "output": "xn--nxasmm1c.com"
+  },
+  {
+    "input": "ශ්‍රී.com",
+    "output": "xn--10cl1a0b660p.com"
+  },
+  {
+    "input": "نامه‌ای.com",
+    "output": "xn--mgba3gch31f060k.com"
+  },
+  {
+    "input": "www.looĸout.net",
+    "output": "www.xn--looout-5bb.net"
+  },
+  {
+    "input": "ᗯᗯᗯ.lookout.net",
+    "output": "xn--1qeaa.lookout.net"
+  },
+  {
+    "input": "www.lookout.сом",
+    "output": "www.lookout.xn--l1adi"
+  },
+  {
+    "input": "www‥lookout.net",
+    "output": null
+  },
+  {
+    "input": "www.lookout‧net",
+    "output": "www.xn--lookoutnet-406e"
+  },
+  {
+    "input": "www.lookout.net⩴80",
+    "output": null
+  },
+  {
+    "input": "www .lookout.net",
+    "output": null
+  },
+  {
+    "input": " lookout.net",
+    "output": null
+  },
+  {
+    "input": "\u001flookout.net",
+    "output": null
+  },
+  {
+    "input": "look۝out.net",
+    "output": null
+  },
+  {
+    "input": "look᠎out.net",
+    "output": null
+  },
+  {
+    "input": "look⁠out.net",
+    "output": "lookout.net"
+  },
+  {
+    "input": "look﻿out.net",
+    "output": "lookout.net"
+  },
+  {
+    "input": "look🿾out.net",
+    "output": null
+  },
+  {
+    "input": "look￺out.net",
+    "output": null
+  },
+  {
+    "input": "look⿰out.net",
+    "output": null
+  },
+  {
+    "input": "looḱout.net",
+    "output": "xn--looout-kp7b.net"
+  },
+  {
+    "input": "look‮out.net",
+    "output": null
+  },
+  {
+    "input": "look⁫out.net",
+    "output": null
+  },
+  {
+    "input": "look󠀁out.net",
+    "output": null
+  },
+  {
+    "input": "look󠀠out.net",
+    "output": null
+  },
+  {
+    "input": "look־out.net",
+    "output": null
+  },
+  {
+    "input": "Bücher.de",
+    "output": "xn--bcher-kva.de"
+  },
+  {
+    "input": "♥.net",
+    "output": "xn--g6h.net"
+  },
+  {
+    "input": "͸.net",
+    "output": null
+  },
+  {
+    "input": "Ӏ.com",
+    "output": null
+  },
+  {
+    "input": "㛼.com",
```

Comment:
This does not fail for jsdom. It gives `xn--snl.com`.

Might be a bug in jsdom, might be a bug in WebKit; haven't yet investigated.

Reply:
That is a Unicode 16 change it seems. I went with the jsdom result even though we haven't updated other tests for Unicode 16 yet. I could also keep it map to null as per Unicode 15 and we update it in 2025 with the others. Up to you.

Reply:
Hmm, I wonder why whatwg-url gets it right, since it claims to be on Unicode 15.1.0.

I slightly prefer updating to Unicode 16, maybe with a note in a nearby README to explain the mixture. But I'm OK either way, since I can't update the library for real until we update the other WPTs.

Reply:
No idea, but it's easy to verify by going to https://www.unicode.org/Public/idna/ and clicking a version then opening `IdnaMappingTable.txt` and searching for `2F868`. I did add an inline comment for this test that it's on Unicode 16.

Reply:
I have looked into this test case. Starting with Unicode 15.1.0, disallowed code points are checked after NFC normalization. And NFC normalizes `U+2F868` to `U+36FC`. So this test case should pass in 15.1.0, 16.0.0 implementations, but will fail in 15.0.0.

Reply:
That's very interesting, though why is it present in the mapping table at all then?